File: cs50x/Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    float letters = 0;
    float words = 1;
    float sentences = 0;
    // string a="Would you like them here or there? I would not like them here or there. I would not
    // like them anywhere."; (Grade 2)
    string a = get_string("Text: ");
    int len = strlen(a);
    for (int n = 0; n < len; n++)
    {
        char c = a[n];
        if (isalpha(c))
        {
            letters++;
        }
        else if (c == ' ')
        {
            words++;
        }
        else if (c == '!' || c == '.' || c == '?')
        {
            sentences++;
        }
    }
    // printf("%i\n%i\n%i\n",letters, words,sentences);

    float L = (letters / (words / 100));
    // printf("%f\n", L);
    float S = (sentences / (words / 100));
    // printf("%f\n%f\n", L,S);
    float end = (0.0588 * L) - (0.296 * S) - 15.8;
    int index = round(end);
    // printf("%i\n",index);
    if (index <= 16 && index >= 1)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: cs50x/Week 6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

word = 1
letter = 0
sentence = 0
for char in text:
    if char == " ":
        word += 1
    elif char == "." or char == "!" or char == "?":
        sentence += 1
    elif char.isalpha():
        letter += 1

L = (letter/word)*100
S = (sentence/word)*100

res = round(0.0588 * L - 0.296 * S - 15.8)
if res < 1:
    print("Before Grade 1")
elif res > 16:
    print("Grade 16+")
else:
    print(f"Grade {res}")
